src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -1589,10 +1589,52 @@ internal IScaleMonitor GetScaleMonitor(string functionId, FunctionName functionN
             {
                 // the durability provider does not support runtime scaling.
                 // Create an empty scale monitor to avoid exceptions (unless runtime scaling is actually turned on).
-                return new NoOpScaleMonitor($"{functionId}-DurableTaskTrigger-{this.Options.HubName}".ToLower());
+                return new NoOpScaleMonitor($"{functionId}-DurableTaskTrigger-{this.Options.HubName}".ToLower(), functionId);
             }
         }
+#endif
+#if FUNCTIONS_V3_OR_GREATER
+
+        internal ITargetScaler GetTargetScaler(string functionId, FunctionName functionName, string connectionName)
+        {
+            if (this.defaultDurabilityProvider.TryGetTargetScaler(
+                    functionId,
+                    functionName.Name,
+                    this.Options.HubName,
+                    connectionName,
+                    out ITargetScaler targetScaler))
+            {
+                return targetScaler;
+            }
+            else
+            {
+                // the durability provider does not support target-based scaling.
+                // Create an empty target scaler to avoid exceptions (unless target-based scaling is actually turned on).
+                return new NoOpTargetScaler(functionId);
+            }
+        }
+
+        private sealed class NoOpTargetScaler : ITargetScaler
+        {
+            /// <summary>
+            /// Construct a placeholder target scaler.
+            /// </summary>
+            /// <param name="functionId">The function ID.</param>
+            public NoOpTargetScaler(string functionId)
+            {
+                this.TargetScalerDescriptor = new TargetScalerDescriptor(functionId);
+            }
+
+            public TargetScalerDescriptor TargetScalerDescriptor { get; private set; }

```

Comment:
Can the `private set` be removed?
```suggestion
            public TargetScalerDescriptor TargetScalerDescriptor { get; }
```

Reply:
Incorporated in my next commit